testinfra/modules/file.py

```diff
@@ -303,3 +307,167 @@ class NetBSDFile(BSDFile):
     def sha256sum(self):
         return self.check_output(
             "cksum -a sha256 < %s", self.path)
+
+
+class WindowsFile(File):
+    @property
+    def exists(self):
+        """Test if file exists
+
+        >>> host.file(r"C:/Users").exists
+        True
+
+        >>> host.file(r"C:/nonexistent").exists
+        False
+        """
+
+        return self.check_output(
+            "powershell -command \"Test-Path %s\"", self.path) == "True"
+
+    @property
+    def is_file(self):
+        return self.check_output(
+            "powershell -command \"(Get-Item %s) -is [System.IO.FileInfo]\"",
+            self.path) == "True"
+
+    @property
+    def is_directory(self):
+        return self.check_output(
+            "powershell -command \"(Get-Item %s) -is [System.IO.DirectoryInfo]\"",  # noqa: E501
+            self.path) == "True"
+
+    @property
+    def is_pipe(self):
+        raise NotImplementedError
+
+    @property
+    def is_socket(self):
+        raise NotImplementedError
+
+    @property
+    def is_symlink(self):
+        raise NotImplementedError
+
+    @property
+    def linked_to(self):
+        """Resolve symlink
+
+        >>> host.file("/var/lock").linked_to
+        '/run/lock'
+        """
+        raise NotImplementedError
+
+    @property
+    def user(self):
+        """Return file owner as string
+
+        >>> host.file("/etc/passwd").user
+        'root'
+        """
+        raise NotImplementedError
+
+    @property
+    def uid(self):
+        """Return file user id as integer
+
+        >>> host.file("/etc/passwd").uid
+        0
+        """
+        raise NotImplementedError
+
+    @property
+    def group(self):
+        raise NotImplementedError
+
+    @property
+    def gid(self):
+        raise NotImplementedError
+
+    @property
+    def mode(self):
+        """Return file mode as octal integer
+
+        >>> host.file("/etc/passwd").mode
+        384  # 0o600 (octal)
+        >>> host.file("/etc/password").mode == 0o600
+        True
+        >>> oct(host.file("/etc/password").mode) == '0600'
+        True
+
+        Note: Python 3 oct(x)_ function will produce ``'0o600'``
+
+        You can also utilize the file mode constants from
+        the stat_ library for testing file mode.
+
+        >>> import stat
+        >>> host.file("/etc/password").mode == stat.S_IRUSR | stat.S_IWUSR
+        True
+
+        .. _oct(x): https://docs.python.org/3.5/library/functions.html#oct
+        .. _stat: https://docs.python.org/2/library/stat.html
+        """
+        raise NotImplementedError
+
+    def contains(self, pattern):
+        return self.run_test("grep -qs -- %s %s", pattern, self.path).rc == 0
+
+    @property
+    def md5sum(self):
+        raise NotImplementedError
+
+    @property
+    def sha256sum(self):
+        raise NotImplementedError
+
+    def _get_content(self, decode):
+        out = self.run_test("cat -- %s", self.path)
```

Comment:
does windows has cat command ?

Reply:
Only in PowerShell. It's an alias for Get-Content
The equivalent in cmd would be type
https://ss64.com/nt/type.html

Reply:
Is this resolved?